launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -336,4 +334,18 @@ static void addPermGenSizeOpt(List<String> cmd) {
     cmd.add("-XX:MaxPermSize=256m");
   }
 
+  /**
+   * Get the major version of the java.version string supplied.
+   */
+  static int javaMajorVersion(String javaVersion) {
+    String[] version = javaVersion.split("[+.\\-]+");
```

Comment:
Ah, yes, you're right that handling + is probably superfluous for java.version. I can prune that from the commit if you want, but doesn't seem to do any harm to leave it in for completeness.